fact_extractor/plugins/unpacking/boschtool/code/boschtool.py

```diff
@@ -0,0 +1,88 @@
+import struct
+from contextlib import suppress
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Dict
+
+from common_helper_process import execute_shell_command
+
+NAME = 'BoschFirmwareTool'
+MIME_PATTERNS = ['firmware/bosch']
+VERSION = '0.1'
+
+TOOL_PATH = Path(__file__).parent.parent / 'bin' / 'boschfwtool'
+
+
+class FirmwareHeader:
+    header_length = 0x400
+    magic_string = b"\x10\x12\x20\x03"
+
+    def __init__(self, file_content: bytes, offset: int = 0, is_subheader: bool = False):
+        self.file_content = file_content
+        self.is_subheader = is_subheader
+        self.offset = offset
+        header_content = file_content[offset:]
+        self.magic = header_content[0:4]
+        self.target = header_content[4:8]
+        self.variant = header_content[8:12]
+        self.version = struct.unpack('>4b', header_content[12:16])
+        self.length = struct.unpack('>i', header_content[16:20])[0]
+        self.base = header_content[20:24]
+        self.checksum = header_content[24:28]
+        self.type = struct.unpack('>i', header_content[28:32])[0]
+        self.signature = header_content[76:332]
+        self.key_blob = header_content[588:844]
+
+    def __str__(self):
+        header_prefix = 'Sub-' if self.is_subheader else ''
+        output = [f'Firmware {header_prefix}Header at offset {self.offset}:']
+        for attribute in ['magic', 'target', 'variant', 'version', 'length', 'base', 'checksum', 'type', 'signature', 'key_blob']:
```

Comment:
`key_blob` and `signature` break the FACT unpacker view, as these are super long strings that are not line-broken.
The question arises if that's up to this plugin or the FACT side (unpacker view) to fix.

Reply:
to me it sounds like some line breaking css black magic should be applied to the template